lib/src/lifecycle_module.dart

```diff
@@ -103,12 +101,33 @@ abstract class LifecycleModule extends SimpleModule
       _didResumeController = new StreamController<LifecycleModule>.broadcast()
     ].forEach(manageStreamController);
 
-    _logger = new Logger('w_module');
+    // The didUnload event must be emitted after disposal which requires that
+    // the stream controller must be disposed of manually at the end of the
+    // unload transition.
+    _didUnloadController = new StreamController<LifecycleModule>.broadcast();
+    _postUnloadDisposable.manageStreamController(_didUnloadController);
+
+    <
+        String,
+        Stream>{
+      'didLoad': didLoad,
+      'didLoadChildModule': didLoadChildModule,
+      'didResume': didResume,
+      'didSuspend': didSuspend,
+      'didUnload': didUnload,
+      'didUnloadChildModule': didUnloadChildModule,
+      'willLoad': willLoad,
+      'willLoadChildModule': willLoadChildModule,
+      'willResume': willResume,
+      'willSuspend': willSuspend,
+      'willUnload': willUnload,
+      'willUnloadChildModule': willUnloadChildModule,
+    }.forEach(_logLifecycleEvents);
   }
 
   /// Name of the module for identification in exceptions and debug messages.
   // ignore: unnecessary_getters_setters
-  String get name => _name;
+  String get name => _name ?? 'LifecycleModule($runtimeType)';
```

Comment:
@georgelesica-wf I updated the default name to use a recognizable string that provides the instance symbol when available.

The release notes should be updated to instruct consumers to override `name` to ensure their log statements always include a recognizable symbol.